apps/condo/domains/user/schema/CheckUserExistenceService.test.js

```diff
@@ -0,0 +1,265 @@
+                })
+                expect(result).toEqual(expectedResult)
+            })
+
+            test('if user is not exist', async () => {
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, { isPhoneVerified: true })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: userType,
+                })
+                expect(result).toEqual({
+                    userIsExist: false,
+                    nameIsSet: false,
+                    emailIsSet: false,
+                    phoneIsSet: false,
+                    passwordIsSet: false,
+                })
+            })
+
+            test('if user was reset', async () => {
+                const [resetUser, resetUserAttrs] = await createTestUser(adminClient, {
+                    type: userType,
+                })
+                await resetUserByTestClient(adminClient, { user: { id: resetUser.id } })
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, {
+                    isPhoneVerified: true,
+                    phone: resetUserAttrs.phone,
+                })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: resetUserAttrs.type,
+                })
+                expect(result).toEqual({
+                    userIsExist: false,
+                    nameIsSet: false,
+                    emailIsSet: false,
+                    phoneIsSet: false,
+                    passwordIsSet: false,
+                })
+            })
+
+            test('if user was soft deleted', async () => {
+                const [deletedUser, deletedUserAttrs] = await createTestUser(adminClient, {
+                    type: userType,
+                })
+                await updateTestUser(adminClient, deletedUser.id, {
+                    deletedAt: new Date().toDateString(),
+                })
+                const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, {
+                    isPhoneVerified: true,
+                    phone: deletedUserAttrs.phone,
+                })
+                const [result] = await checkUserExistenceByTestClient(anonymousClient, {
+                    confirmActionToken: confirmPhoneAction.token,
+                    userType: deletedUserAttrs.type,
+                })
+                expect(result).toEqual({
+                    userIsExist: false,
+                    nameIsSet: false,
+                    emailIsSet: false,
+                    phoneIsSet: false,
+                    passwordIsSet: false,
+                })
+            })
+        })
+
+        test('confirmPhoneAction should not be marked as used after checkUserExistence has been executed', async () => {
```

Comment:
does it mean, that I can reuse one confirmPhoneAction to trigger only this handle?

Reply:
with one `confirmPhoneAction` you can call `checkUserExistence` and then something else, like `signInUser`

Reply:
I still not quite sure this one is right...

Reply:
But can I use one confirmPhoneAction to call this particular query? Maybe from different IPs (VPNs and such)

Reply:
> But can I use one confirmPhoneAction to call this particular query? Maybe from different IPs (VPNs and such)

You can.
Similarly, you can call query "getPhoneByConfirmPhoneActionToken"

Reply:
> I still not quite sure this one is right...

We shouldn't ask the user to confirm their phone number 10 times to register, only 1... So when using a token in this query, we don't mark it as used

In this case we can allow it